src/components/button/button.tsx

```diff
@@ -0,0 +1,93 @@
+import { ButtonKind, ButtonSize } from 'boilerplate-react-native/src/types/button';
+import React from 'react';
+import {
+  TouchableOpacity,
+  Text,
+  ActivityIndicator,
+  View,
+  GestureResponderEvent,
+} from 'react-native';
+
+import { useButtonStyles, useKindStyles, useSizeStyles } from './button.styles';
+interface ButtonProps {
+  disabled?: boolean;
+  endEnhancer?: React.ReactElement | string;
+  isLoading?: boolean;
+  kind?: ButtonKind;
+  onClick?: (event: GestureResponderEvent) => void;
+  size?: ButtonSize;
+  startEnhancer?: React.ReactElement | string;
+  children: React.ReactNode;
```

Comment:
No need for this.

You can update component signature as `const Button: React.FC<PropsWithChildren<ButtonProps>>`